### PR TITLE
WIP - Feature/op 2.0

### PR DESCRIPTION
**Pull request by Dvelezs94**

### CODE REVIEW

**Comment by Dvelezs94** [comment]
This is still under development. will leave the PR open to welcome ideas and changes. There are still errors when authenticating against 1password on mac, so I'm fixing that

**Comment by dtpryce** [comment]
Thanks @Dvelezs94 for the submission! Keep going and once you're happy message here, I won't be able to help much currently but if I (or one of my team) get time we will tell you and see where we can help best :D

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.6.2
+0.6.3
```
Comment (dtpryce) [inline VERSION:1]: Fairly certain that op2 isn't backwards compatible ... can you bump this to 1.0.0?

**Comment by Dvelezs94** [comment]
@dtpryce login is working fine now, however I messed something up. Basically the OP_SESSSION_xxx is being set on bashrc, however it needs to be loaded on every execution of `op <resource> <action>`, or to be set as environment variable.

**Comment by Dvelezs94** [comment]
closed by mistake...

**Comment by dtpryce** [comment]
> closed by mistake...
> 
> Still working on it. OP_SESSION_account is working on both mac and linux, however I'm still missing a few things
> 
>     * Make it import the OP_SESSION_account on action execution
> 
>     * Make it load existing OP_SESSION on subsequent runs

No worries at all @Dvelezs94 just appreciate you doing this! 

One thought I had when I looked at op2 a while back was that it might not need the bash profile handling for subsequent runs but I can't remember why. Any thoughts on this or was I just wrong?

**Comment by Dvelezs94** [comment]
> > closed by mistake...
> > Still working on it. OP_SESSION_account is working on both mac and linux, however I'm still missing a few things
> > ```
> > * Make it import the OP_SESSION_account on action execution
> > 
> > * Make it load existing OP_SESSION on subsequent runs
> > ```
> 
> No worries at all @Dvelezs94 just appreciate you doing this!
> 
> One thought I had when I looked at op2 a while back was that it might not need the bash profile handling for subsequent runs but I can't remember why. Any thoughts on this or was I just wrong?

we need to save the OP_SESSION_<account> in the bash profile because the `op` command looks for the environment variable, however we need to import it on signin.. thats what Im working on.

A question I have is why do we encrypt the password?

**Comment by dtpryce** [comment]
> > > closed by mistake...
> > > Still working on it. OP_SESSION_account is working on both mac and linux, however I'm still missing a few things
> > > ```
> > > * Make it import the OP_SESSION_account on action execution
> > > 
> > > * Make it load existing OP_SESSION on subsequent runs
> > > ```
> > 
> > 
> > No worries at all @Dvelezs94 just appreciate you doing this!
> > One thought I had when I looked at op2 a while back was that it might not need the bash profile handling for subsequent runs but I can't remember why. Any thoughts on this or was I just wrong?
> 
> we need to save the OP_SESSION_ in the bash profile because the `op` command looks for the environment variable, however we need to import it on signin.. thats what Im working on.
> 
> A question I have is why do we encrypt the password?

Great question, we initially did this since we are saving and transferring passwords from kernel to disc and wanted to be as secure as possible. Now sadly we have a CVE around storing them in plain text https://github.com/wandera/1password-client/security/code-scanning/1 which we should try to fix too but not entirely sure how to without digging in.

**Comment by dtpryce** [comment]
@Dvelezs94 it looks like we are going to need this since our org is moving 1Password onto SSO which is supported by ... you guessed it ... cli v2! We will include your awesome work in #40 and feel free to work with us. The part we have got up to is the new login workflow that except one edge case (no local 1P app installed, linked with CLI) users will not need passwords anymore which is very interesting.

**Comment by dtpryce** [comment]
Hey @Dvelezs94 we have just released 1.0.1 thanks to your contribution and we now fully support v2 of the cli and app. Please update from pypi and make sure to follow the cli setup instructions to link to your app etc.

**Pull request closed without merging**